cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb

```diff
@@ -0,0 +1,1143 @@
+        "# limitations under the License."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "C5SwRvvKJvcz"
+      },
+      "source": [
+        "# **Building AI-powered data-driven applications using pgvector, LangChain and LLMs**\n",
+        "\n",
+        "---"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "27wCSEVJhmjp"
+      },
+      "source": [
+        "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/GoogleCloudPlatform/python-docs-samples/blob/main/cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_GhFgSOldCvV"
+      },
+      "source": [
+        "This hands-on tutorial will show you how you can add Generative AI features to your own applications with just a few lines of code using pgvector, LangChain and LLMs on Google Cloud.\n",
+        "\n",
+        "We will build together a sample Python application that will be able to understand and respond to human language queries about the relational data stored in your PostgreSQL database. In fact, we will further push the creative limits of the application by teaching it to generate new content based on our existing dataset.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "93QUUCrvKm03"
+      },
+      "source": [
+        "## Objective\n",
+        "\n",
+        "After completing the steps in this notebook:\n",
+        "- You will have a good understanding of how to use the [pgvector extension](https://github.com/pgvector/pgvector) to store and search vector embeddings in PostgreSQL. Learn more about [vector embeddings](https://cloud.google.com/blog/topics/developers-practitioners/meet-ais-multitool-vector-embeddings).\n",
+        "- You will get a hands-on experience with using the open-source [LangChain framework](https://python.langchain.com/en/latest/index.html) to develop applications powered by large language models. LangChain makes it easier to develop and deploy applications against any LLM model in a vendor-agnostic manner.\n",
+        "- You will learn about the powerful features in [Google PaLM models made available through Vertex AI](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/**overview**).\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "DS7GdlJ1XowY"
+      },
+      "source": [
+        "## Example Scenario\n",
+        "\n",
+        "This notebook uses an example of an e-commerce company that runs an online marketplace for buying and selling children toys (it is a \"toy\" example, after all 😋). This company wants to add new Generative AI experiences in their e-commerce applications for both buyers and sellers on their platform.\n",
```

Comment:
nit: omit `(it is a \"toy\" example, after all 😋)`

Reply:
Removed.